src/app/AccessPrivilegeGet.cpp

```diff
@@ -0,0 +1,38 @@
+/*
```

Comment:
This needs a better filename, which would indicate that these are here just to be overridden.  `DefaultRestrictivePrivilegeGetters` or something like that, perhaps?

But I have to ask: why do we need these at all?  Why aren't all consumers just providing these functions, so we can avoid weak symbols altogether?

Reply:
Changed the filename.

As described in #16540, RequiredPrivilege is built into the DM/IM library, but the underlying data is built per app/example. If we remove the weak functions there will be compilation errors for the chip-im-responder/initiator. But if we place the weak functions in `RequiredPrivilege.cpp`, the functions in `src/app/utils/privilege-storage.cpp` might not  be able to override the weak functions for the apps/examples. 
We only called the `MatterGetAccessPrivilegeForXX` functions in `RequiredPrivilege.h` which is included by `RequiredPrivilege.cpp`, so the linker might ignore the override functions and use the weak functions intead.

Reply:
None of that answers my question: why aren't all consumers providing these functions?  That was the initial intent; unfortunately @mlepage-google has not been able to get to it.  But we should implement that initial intent instead of adding more workarounds for the fundamental weak-function brokenness....

Reply:
Created another PR #21220 removing the weak functions and adding the stubs for `test:helper` & `chip-im-responder/initiator`.